src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -498,6 +531,23 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 
 	if (ShouldSyncTableMetadata(relationId))
 	{
+		if (ClusterHasKnownMetadataWorkers())
+		{
+			/*
+			 * Ensure sequence dependencies and mark them as distributed
+			 * before creating table metadata on workers
+			 */
+			Oid sequenceOid = InvalidOid;
+			foreach_oid(sequenceOid, dependentSequenceList)
+			{
+				/* get sequence address */
+				ObjectAddress sequenceAddress = { 0 };
+				ObjectAddressSet(sequenceAddress, RelationRelationId, sequenceOid);
+				EnsureDependenciesExistOnAllNodes(&sequenceAddress);
+				MarkObjectDistributed(&sequenceAddress);
```

Comment:
Wondering if these 4 lines can move into a separate function, since they recur a lot.

Reply:
Makes sense to write separate functions. I wanted to make sure that their positioning makes sense first.